File: advent4.go
package main

import (
	"fmt"
	"crypto/md5"
	"encoding/hex"
	//"strings"
	//"strconv"
)

func main() {

	const input = "yzbqklnj"
	hasher := md5.New()
	hasher.Write([]byte(input))

	fmt.Printf(hex.EncodeToString(hasher.Sum(nil)))


	//fmt.Printf("%x", strings.Contains(hash, "dd"))
}

	

File: advent4.py
import hashlib

input = "yzbqklnj"

m = hashlib.md5()
m.update(input)

found = False
count = -1
next = input

while not found:
	count += 1
	next = input + str(count)
	m = hashlib.md5()
	m.update(next)
	if m.hexdigest()[:6] == "000000":
		found = True
		print m.hexdigest()
		print count
